### ISSUE TITLE
Battery Plugin not working

### ISSUE THREAD

**Issue by KevinYobeth** [flutter/flutter#33517]
I follow all the guide on the https://pub.dev/packages/battery page, but it throw an error

E/flutter (10474): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: MissingPluginException(No implementation found for method getBatteryLevel on channel plugins.flutter.io/battery)
E/flutter (10474): #0      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:300:7)
E/flutter (10474): <asynchronous suspension>
E/flutter (10474): #1      Battery.batteryLevel (package:battery/battery.dart:36:8)
E/flutter (10474): #2      mainAppState.check (package:ui/main.dart:17:25)
E/flutter (10474): <asynchronous suspension>
E/flutter (10474): #3      mainAppState.change (package:ui/main.dart:21:5)
E/flutter (10474): #4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:511:14)
E/flutter (10474): #5      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:566:30)
E/flutter (10474): #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:166:24)
E/flutter (10474): #7      TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:240:9)
E/flutter (10474): #8      TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:177:9)
E/flutter (10474): #9      PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:436:9)
E/flutter (10474): #10     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
E/flutter (10474): #11     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
E/flutter (10474): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:221:19)
E/flutter (10474): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:199:22)
E/flutter (10474): #14     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:156:7)
E/flutter (10474): #15     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:102:7)
E/flutter (10474): #16     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:86:7)
E/flutter (10474): #17     _rootRunUnary (dart:async/zone.dart:1136:13)
E/flutter (10474): #18     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter (10474): #19     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
E/flutter (10474): #20     _invoke1 (dart:ui/hooks.dart:233:10)
E/flutter (10474): #21     _dispatchPointerDataPacket (dart:ui/hooks.dart:154:5)

**Comment by ketanm** [comment]
Hi KevinYobeth,

A couple of questions:

- Was error thrown on Simulator or actual platform? Also, what are your host (development OS) and target platform?

- Did you run `flutter doctor` and `flutter package get`? If you can share the output of `flutter doctor` it will be helpful to debug and help.

**Comment by tomosullivan8** [comment]
Hi, i'm getting a similar issue when using the battery package. I get the message;

E/flutter ( 4856): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: PlatformException(UNAVAILABLE, Charging status unavailable, null)
E/flutter ( 4856): null

Anyone know a fix for this??

I've run the flutter doctor and everything is fine.

**Comment by tomosullivan8** [comment]
@ketanm also it's both the simulator and on an android device. I've just run the package example from scratch to see if it's any implementation but get the same error message.

**Comment by ketanm** [comment]
Hi,

I just pulled the package to latest code, ran `flutter build apk` and installed `app-release.apk` on my Android phone and ran the application successfully. 

Can you give more details about your setup and how you are building?

Ketan

**Comment by iapicca** [comment]
Hi @KevinYobeth 
are you still experiencing this issue with the latest version of Flutter?
If so please reply attaching your flutter doctor -v.
Also, to better address the issue, would be helpful
if you could post a self contained app on github
or the steps to reproduce it.
Thank you

**Comment by iapicca** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!
  
Thanks for your contribution.

**Comment by begoon** [comment]
Still having this issue.

I'm building the demo application https://github.com/flutter/plugins/blob/master/packages/battery/battery/example/lib/main.dart.

```
> flutter run                                                                                                            
Running "flutter pub get" in battery...                             1.0s
Launching lib/main.dart on A920 in debug mode...
Running Gradle task 'assembleDebug'...
Running Gradle task 'assembleDebug'... Done                        16.6s
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...                15.2s
Waiting for A920 to report its views...                             12ms
Syncing files to device A920...                                    552ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on A920 is available at: http://127.0.0.1:57925/MuhkkiG3zKU=/
E/flutter (23346): [ERROR:flutter/lib/ui/ui_dart_state.cc(166)] Unhandled Exception: PlatformException(UNAVAILABLE, Charging status unavailable, null)
E/flutter (23346):
```

```
> flutter doctor -v                                                                                                       ~/github/battery 0
[✓] Flutter (Channel stable, 1.20.4, on Mac OS X 10.15.6 19G2021, locale en-GB)
    • Flutter version 1.20.4 at /Users/alexander/flutter
    • Framework revision fba99f6cf9 (4 days ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/alexander/Library/Android/sdk
    • Platform android-29, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0, Build version 12A7209
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[✓] Connected device (1 available)
    • A920 (mobile) • 0821249013 • android-arm • Android 7.1.2 (API 25)

• No issues found!```

**Comment by begoon** [comment]
The issue will be fixed by https://github.com/flutter/flutter/issues/66125.
